libraries/AP_Compass/AP_Compass_UAVCAN.cpp

```diff
@@ -204,4 +221,67 @@ void AP_Compass_UAVCAN::read(void)
     drain_accumulated_samples(_instance);
 }
 
+#if AP_TEST_DRONECAN_DRIVERS
+void AP_Compass_UAVCAN::_setup_eliptical_correcion(uint8_t i)
```

Comment:
I find it strange to have this test code here. Why not in libraries/SITL/ ?

Reply:
Its just replicating, AP_Compass_SITL.cpp

Reply:
I have moved all such tests out of drivers and into libraries/SITL as discussed.